Sources/Common/DataModel/DataSet/Constants.d.ts

```diff
@@ -0,0 +1,10 @@
+export declare enum FieldAssociations {
+	FIELD_ASSOCIATION_POINTS = 0,
+	FIELD_ASSOCIATION_CELLS = 1,
+	FIELD_ASSOCIATION_NONE = 2,
+	FIELD_ASSOCIATION_POINTS_THEN_CELLS = 3,
+	FIELD_ASSOCIATION_VERTICES = 4,
+	FIELD_ASSOCIATION_EDGES = 5,
+	FIELD_ASSOCIATION_ROWS = 6,
+	NUMBER_OF_ASSOCIATIONS = 7,
+}
```

Comment:
maybe we also need FieldDataTypes from Constants.js

Reply:
also we need to export them in index.d.ts

```typescript
export declare const vtkDataSet: {
	newInstance: typeof newInstance;
	extend: typeof extend;
    FieldAssociation: typeof FieldAssociation;
    FieldDataTypes: typeof FieldDataTypes;
}; 
```

Reply:
`FieldDataTypes` has some redundant enum value. (eg `COORDINATE` = `POINT_DATA`). Don't know if it's gonna play well with TypeScript.

Reply:
That should be fine. Two distinct enum names can share the same value, so long as its specified.